lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -91,6 +92,31 @@ sub view ($self) {
     $self->viewimg;
 }
 
+sub _create_tmpdir_for_needles_refspec ($self, $job) {
+    return undef unless $self->app->config->{'scm git'}->{checkout_needles_sha} eq 'yes';
+    my $needle_dir = $job->needle_dir;
+    $needle_dir = realpath($needle_dir) // $needle_dir;
+    my $needles_dir_var = $job->settings->single({key => 'NEEDLES_DIR'});
+    $needles_dir_var = $job->settings->single({key => 'CASEDIR'}) unless $needles_dir_var;
+    return undef unless $needles_dir_var;
+    my $needles_url = Mojo::URL->new($needles_dir_var->value);
+    return undef unless $needles_url->scheme;
+    my $needles_ref = $needles_url->fragment;
+    eval {
+        my $vars_json = Mojo::File->new($job->result_dir(), 'vars.json')->slurp;
+        my $vars = decode_json($vars_json);
+        $needles_ref = $vars->{NEEDLES_GIT_HASH};
+    };
+    chomp($needles_ref);
+    return undef unless $needles_ref;
+    qx{git -C "$needle_dir" fetch --depth 1 origin "$needles_ref" &>/dev/null};
+    $needles_ref = qx{git -C "$needle_dir" rev-parse FETCH_HEAD};
+    my $needle_dir_basename = basename(dirname($needle_dir));
+    my $new_path = "/tmp/needle_dirs/$needle_dir_basename/$needles_ref/needles";
+    make_path($new_path);
```

Comment:
The error handling with `File::Path::make_path` is rather difficult: https://metacpan.org/pod/File::Path#ERROR-HANDLING

I think we shouldn't just log to stderr here (which would be the default). Maybe it is better to use `Mojo::File::make_path` instead and handle the exception it might throw as needed.